src/engraving/libmscore/chordrest.cpp

```diff
@@ -408,6 +409,25 @@ void ChordRest::undoSetSmall(bool val)
     undoChangeProperty(Pid::SMALL, val);
 }
 
+//---------------------------------------------------------
+//   trimLength - shorten a spanner so it ends on a note/chord
+//---------------------------------------------------------
+
+void trimTrailingRests(Spanner* spanner, ChordRest* chordRest) {
+    Fraction length = spanner->ticks();
+    Fraction trimTo;
+    while (chordRest = Ms::nextChordRest(chordRest, true)) {
+        auto crTicks = chordRest->ticks();
+        if (!chordRest->isRest()) {
+            trimTo += crTicks;
+        }
+        if ((length -= crTicks) <= Fraction()) {
+            break;
+        };
+    }
+    spanner->setTicks(trimTo);
```

Comment:
should probably look further to see what existing functions (similar to findCR etc.) might do this